Move CNCF Kubernetes to new provider structure

jason810496 opened the pull request:
related: #46045 



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

jason810496 committed:
Move CNCF Kubernetes to new provider structure

jason810496 committed:
Fix doc include path and k8s test

jason810496 committed:
Fix taskflow tutorial

jason810496 committed:
Fix test_project_structure

rawwar commented on tests/always/test_project_structure.py:
```diff
@@ -248,6 +248,8 @@ def list_of_classes(self):
             operators_paths = self.get_classes_from_file(operator_file, NEW_PROVIDER_SRC, is_new=True)
             classes.update(operators_paths)
+        # replace src. prefix with empty string for all classes
+        classes = {k.replace("src.", ""): v for k, v in classes.items()}
```
```suggestion
        # replace src. prefix with empty string for all classes
        classes = {k.strip("src."): v for k, v in classes.items()}
```

rawwar replied:
we don't want to replace all "src." occurrences

jason810496 committed:
Strip src. prefix instead of replacing all src.

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

rawwar commented on tests/always/test_project_structure.py:
```diff
@@ -571,6 +573,19 @@ class TestCncfProviderProjectStructure(ExampleCoverageTest):
+        "airflow.providers.cncf.kubernetes.operators.resource.KubernetesResourceBaseOperator",
+        "airflow.providers.cncf.kubernetes.operators.spark_kubernetes.SparkKubernetesOperator",
+        "airflow.providers.cncf.kubernetes.sensors.spark_kubernetes.SparkKubernetesSensor",
+    }
```
```suggestion
```

We don't need this now?

rawwar replied:
These still need to be removed and provider name has to be updated to cncf/kubernetes for these to pass. As of now, since we are ignoring these, tests are passing

jason810496 replied:
Thanks for the reminder, just test as you said locally and passed. I will commit right away !

jason810496 committed:
Merge fix get_classes_from_file #46454

jason810496 committed:
Fix TestCncfProviderProjectStructure
- rename PROVIDER from "cncf" to "cncf/kubernetes"
- remove MISSING_EXAMPLES_FOR_CLASSES

jason810496 committed:
Fix k8s CI requirements

potiuk committed:
fixup! Fix k8s CI requirements
